MFCSample/MicroserviceManager/MicroserviceManagerDlg.cpp

```diff
@@ -0,0 +1,338 @@
+	ON_CONTROL_RANGE(BN_CLICKED, IDC_OUT1, IDC_IN2, &CMicroserviceManagerDlg::OnRadioButtonClicked)
+END_MESSAGE_MAP()
+
+
+// CMicroserviceManagerDlg message handlers
+
+BOOL CMicroserviceManagerDlg::OnInitDialog()
+{
+	CDialogEx::OnInitDialog();
+
+	// Add "About..." menu item to system menu.
+
+	// IDM_ABOUTBOX must be in the system command range.
+	ASSERT((IDM_ABOUTBOX & 0xFFF0) == IDM_ABOUTBOX);
+	ASSERT(IDM_ABOUTBOX < 0xF000);
+
+	CMenu* pSysMenu = GetSystemMenu(FALSE);
+	if (pSysMenu != nullptr)
+	{
+		BOOL bNameValid;
+		CString strAboutMenu;
+		bNameValid = strAboutMenu.LoadString(IDS_ABOUTBOX);
+		ASSERT(bNameValid);
+		if (!strAboutMenu.IsEmpty())
+		{
+			pSysMenu->AppendMenu(MF_SEPARATOR);
+			pSysMenu->AppendMenu(MF_STRING, IDM_ABOUTBOX, strAboutMenu);
+		}
+	}
+
+	// Set the icon for this dialog.  The framework does this automatically
+	//  when the application's main window is not a dialog
+	SetIcon(m_hIcon, TRUE);			// Set big icon
+	SetIcon(m_hIcon, FALSE);		// Set small icon
+
+	// TODO: Add extra initialization here
+
+	return TRUE;  // return TRUE  unless you set the focus to a control
+}
+
+void CMicroserviceManagerDlg::OnSysCommand(UINT nID, LPARAM lParam)
+{
+	if ((nID & 0xFFF0) == IDM_ABOUTBOX)
+	{
+		CAboutDlg dlgAbout;
+		dlgAbout.DoModal();
+	}
+	else
+	{
+		CDialogEx::OnSysCommand(nID, lParam);
+	}
+}
+
+// If you add a minimize button to your dialog, you will need the code below
+//  to draw the icon.  For MFC applications using the document/view model,
+//  this is automatically done for you by the framework.
+
+void CMicroserviceManagerDlg::OnPaint()
+{
+	if (IsIconic())
+	{
+		CPaintDC dc(this); // device context for painting
+
+		SendMessage(WM_ICONERASEBKGND, reinterpret_cast<WPARAM>(dc.GetSafeHdc()), 0);
+
+		// Center icon in client rectangle
+		int cxIcon = GetSystemMetrics(SM_CXICON);
+		int cyIcon = GetSystemMetrics(SM_CYICON);
+		CRect rect;
+		GetClientRect(&rect);
+		int x = (rect.Width() - cxIcon + 1) / 2;
+		int y = (rect.Height() - cyIcon + 1) / 2;
+
+		// Draw the icon
+		dc.DrawIcon(x, y, m_hIcon);
+	}
+	else
+	{
+		CDialogEx::OnPaint();
+	}
+}
+
+// The system calls this function to obtain the cursor to display while the user drags
+//  the minimized window.
+HCURSOR CMicroserviceManagerDlg::OnQueryDragIcon()
+{
+	return static_cast<HCURSOR>(m_hIcon);
+}
+
+
+Json::Value jsonDeviceState;
+
+std::map<std::string, Json::Value> jsonToMap(const Json::Value& jsonValue) {
+	std::map<std::string, Json::Value> resultMap;
+
+	// Iterate through the JSON object
+	for (const auto& key : jsonValue.getMemberNames()) {
+		resultMap[key] = jsonValue[key];
+	}
+
+	return resultMap;
+}
+
+void CMicroserviceManagerDlg::SetSwitchRequest(string selectedDevice, int switchNum, string state)
+{
+	ServiceClient* srv = new ServiceClient("ServiceCleware", "127.0.0.1", 5672, "ServiceClewareKey");
+	Json::Value jsonObject;
+	jsonObject[METHOD] = REQ_SET_SWITCH;
+
+	Json::Value listArray;
```

Comment:
Hi Cuong,
is there no easier way to create an anonymous data structure on the fly directly in the interface of RequestService?
Thank you,
 Thomas

Reply:
Hello Thomas,

I'm preparing a better design. 

I have many ideas about design, and I tried to implement them, but in the end, I got overwhelmed by a significant number of bugs. At the last minute, I decided to go for the simplest approach to create a running demo.

Thanks,
Cuong

Reply:
Hi Cuong,
no problem. I wanted to give only my ideas. Nice, when you work already on both findings.

Best would be a as simple as possible transparent RPC call provided by generic functions. It came into my mind that maybe also macros could help here. I made long no c++ code :-)

Thank you,
 Thomas